### ISSUE TITLE
Incorrect response from VIP visit

### ISSUE THREAD

**Issue by MarkL1961** [CorsixTH/CorsixTH#835]
Taken this from the forum
http://i.imgur.com/iNZPJBD.png
http://forums.corsixth.com/index.php/topic,5608.msg8097.html?PHPSESSID=a70ecbc699a25878d9fc65155360ae57#new

**Comment by Alberth289346** [comment]
In Lua/entites/vip.lua, around line 151, the mediocre rating (3 of 1..5) fax is constructed, which has a 'mediocre' string, a boost, and a cash award. That looks like our guy.
I cross-checked with the message text, and indeed S[44][203] is a mediocre remark in the original strings. In other words, it's a mediocre VIP rating.

Line 473 defines the cash_reward, and a little below the rating is defines, which is between 45 and 64 for mediocre (probably of 100).

I agree the message is double.

The lesser levels (rating 1 and 2) have a reputation loss and no money, the higher levels (rating 4 and 5) are like the mediocre level at the moment, except for rating 5, which also seems to have a "thank you" action of some kind.

Something "in-between" would be nice, but how to achieve that?
* Drop the boost
* Drop the reward
* Reduce the reward

I don't know what to do here.

**Comment by MarkL1961** [comment]
For such a comment I would think no reward and the reputation in community should go down and the fax should reflect that the reputation is reduced.

**Comment by Alberth289346** [comment]
that's what "bad" and "very bad" do currently, so instead of 3 good-ish ones and 2 bad-ish ones, we'd have 2 good-ish ones and 3 bad-ish ones. I think mediocre deserves something in-between good and bad

Maybe the string itself is wrong?

**Comment by MarkL1961** [comment]
I'll take a look at the strings then and get back to you as I was thinking that one was bad

**Comment by MarkL1961** [comment]
The good ones should get a cash reward and a boost to the rep, but it only tells you about the cash.
Medium get a boost to the rep only and the bad get nothing

**Comment by MarkL1961** [comment]
When john added the VIP I think he did some scheme to work from, I'll see if I can track it down

**Comment by Alberth289346** [comment]
FYI, the string at the fax is https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Lua/languages/original_strings.lua#L2065 Around it you can see the string numbers for the other ratings

**Comment by MarkL1961** [comment]
https://www.dropbox.com/s/p7c40gdjsr7lpet/vip-algorithm.pdf?dl=0

**Comment by MarkL1961** [comment]
Perhaps Super and good get both cash and the boost, the middle and next one get rep boost only and the worst gets rep loss?

**Comment by Alberth289346** [comment]
I made some changes to what seems useful, please let me know what you think.
The table below lists the current implementation (rep-loss = reputation loss, rep-gain = reputation boost, mesg X = you get a message about X).

First table is the current implementation, second table is my proposal, with the differences to the current implementation in italic font (if all is well).


*Current implementation:*

 Field | Very bad | Bad | Average | Good | Very good
--------|------------|-------|-----------|---------|--------------
**reputation** | rep-loss (-10) | rep-loss (-5) | rep-gain | rep-gain | rep-gain
**message reputation** | mesg rep-loss | mesg rep-loss | mesg rep-gain | mesg rep-gain | mesg rep-gain
**money** | nothing  | nothing | money | money | money
**message money** | nothing  | nothing | mesg money | mesg money | mesg money

*Proposal:*

 Field | Very bad | Bad | Average | Good | Very good
--------|------------|-------|-----------|---------|--------------
**reputation** | rep-loss (-10) | *nothing* | rep-gain | rep-gain | rep-gain
**message reputation** | mesg rep-loss | *nothing* | *nothing* | mesg rep-gain | mesg rep-gain
**money** | nothing  | nothing | *nothing* | money | money
**message money** | nothing  | nothing | *nothing* | mesg money | mesg money

**Comment by MarkL1961** [comment]
I'd be inclined to only have the middle one (average) with nothing, not even a rep gain and leave the others as they are
